src/components/ModalBackupWords.js

```diff
@@ -96,59 +105,78 @@ class ModalBackupWords extends React.Component {
         this.props.updateWords(words);
         this.setState({ passwordSuccess: true, errorMessage: '' });
       } else {
-        this.setState({errorMessage: t`Invalid password`})
+        this.setState({ errorMessage: t`Invalid password` });
       }
     }
-  }
+  };
 
   /**
-   * Called when user clicks the button saying that he has saved the words already  
-   * So we select some words to do the validation and show validation component in the modal
+   * Called when user clicks the button saying that he has saved the words already
+   * Then, validation steps will ask the right word given an index between [1,24].
+   * As the words were shown previosly, the user should be able to find any word given an index.
    */
   handleWordsSaved = () => {
-    const wordsValidation = _.shuffle(this.props.words.split(' ')).slice(0, WORDS_VALIDATION);
-    this.setState({ showValidation: true, wordsValidation });
-  }
+    const wordsArray = this.props.words.split(' ');
+    const wordsToValidate = _.shuffle(wordsArray).slice(0, WORDS_VALIDATION);
+    const validationSteps = wordsToValidate.map((word, index) => {
+      const backupIndex = wordsArray.indexOf(word);
+      let optionsStartIndex = backupIndex - 2;
+      let optionsEndIndex = backupIndex + 2;
+
+      // If index is 0 or 1, startIndex would be negative
+      // So we set start to 0 and end to 4
+      if (optionsStartIndex < 0) {
+        optionsStartIndex = 0;
+        optionsEndIndex = optionsStartIndex + 4;
+      }
+
+      // If index is 22 or 23, endIndex would be greater than the max
+      // So we set to the max and decrease the startIndex
+      const maxIndex = wordsArray.length - 1;
+      if (optionsEndIndex > maxIndex) {
+        optionsEndIndex = maxIndex;
+        optionsStartIndex = maxIndex - 4;
+      }
+
+      const options = _.shuffle(
+        wordsArray.slice(optionsStartIndex, optionsEndIndex + 1)
+      );
+
+      return {
+        backupIndex: backupIndex + 1,
+        correctWord: word,
+        options,
+        done: false,
+        last: index === wordsToValidate.length - 1,
+      };
+    });
+    this.setState({ showValidation: true, validationSteps });
+  };
 
   /**
    * Called when user finishes the backup. We validate the backup and shows a success message, in case of success
    */
-  handleValidateBackup = () => {
-    if (this.state.chosenWords.length !== WORDS_VALIDATION) {
-      this.setState({ errorMessage: t`Invalid number of words` });
+  handleValidationStepOption = (option, validationStep) => {
+    validationStep.chosenOption = option;
+    if (option !== validationStep.correctWord) {
+      this.setState({
+        validationSteps: this.state.validationSteps,
```

Comment:
What's this `setState` is doing? It's just saving the same data, isn't it?

Reply:
At line 160 the statement `validationStep.chosenOption = option;` changed the property `chosenOption` of `validationStep`, which is an element of `validationSteps` array. So, `validationSteps` isn't "deep equal" to its previous value before line 160. As this change will only be visible with a re-render, I included `validationSteps` inside the `setState` call to indicate that it has changed, although it is not really necessary and even a `this.forceUpdate()` could do the re-render job. I can do something like the below to be more clear about the intention to change `validationSteps`.
```js
this.setState({
    validationSteps: [ ...this.state.validationSteps ],
    ...
});
``` 
The point here is the need for a re-render to update any markup affected by `validationSteps`. I will do `validationSteps: [ ...this.state.validationSteps ]` clearly indicate that `validationSteps` has changed, ok?

Reply:
Resolved by https://github.com/HathorNetwork/hathor-wallet/pull/156/commits/e3958cb3d1836dbfe98dd12e9cb4c1e60cdfe566.